tests/nested/manual/remodel-cross-store/task.yaml

```diff
@@ -0,0 +1,122 @@
+  SNAPD_TEST_BRAND: test-snapd
+  # store we are remodeling from
+  SNAPD_STORE_WHITEBOX: CQvkdMjgbapk821Po2cF
+  # store we are remodeling to
+  SNAPD_STORE_REMODEL: Ceith6aiSahditeoyizi
+  # device serial, this should match the sub-store mapping rules in the serial
+  # vault. Since this test and remodel-simple are using the same original model,
+  # due to limitations of the snap-store/serial-vault, the serial used in this
+  # test must be different than in remodel-simple.
+  SERIAL: 1234-snapd-remodel-testing-cross-store
+
+prepare: |
+    # shellcheck source=tests/lib/nested.sh
+    . "$TESTSLIB"/nested.sh
+    UBUNTU_IMAGE="$GOHOME"/bin/ubuntu-image
+    variant=18
+    if os.query is-focal; then
+        variant=20
+    fi
+    IMAGE_NAME="$(nested_get_image_name core)"
+    # fetch the snapd snap from edge and repack the deb into the snap
+    snap download --channel="latest/edge" snapd
+    "$TESTSTOOLS"/snaps-state repack_snapd_deb_into_snap snapd
+
+    # TODO: once the gadget snap is published, this will download it from the
+    # store
+    cp -av "$TESTSLIB/snaps/test-snapd-remodel-pc-$variant" test-snapd-remodel-pc
+    test -n "$SERIAL"
+    echo "$SERIAL" > test-snapd-remodel-pc/serial
+    snap pack test-snapd-remodel-pc
+
+    UBUNTU_STORE_AUTH_DATA_FILENAME="$TESTSLIB/remodel-store-viewer.auth" \
+        "$UBUNTU_IMAGE" snap "$TESTSLIB/assertions/test-snapd-remodel-pc-$variant.model" \
+            --snap test-snapd-remodel-pc_*.snap \
+            --snap "$PWD/snapd-from-deb.snap" \
+            --output-dir "$NESTED_IMAGES_DIR"
+    # ubuntu-image dropped the --output parameter, so we have to rename
+    # the image ourselves, the images are named after volumes listed in
+    # gadget.yaml
+    find "$NESTED_IMAGES_DIR/" -maxdepth 1 -name '*.img' | while read -r imgname; do
+        if [ -e "$NESTED_IMAGES_DIR/$IMAGE_NAME" ]; then
+            echo "Image $IMAGE_NAME file already present"
+            exit 1
+        fi
+        mv "$imgname" "$NESTED_IMAGES_DIR/$IMAGE_NAME"
+    done
+
+    nested_create_assertions_disk
+    nested_start_core_vm
+
+restore: |
+    # shellcheck source=tests/lib/nested.sh
+    . "$TESTSLIB"/nested.sh
+    nested_shutdown
+
+execute: |
+    # shellcheck source=tests/lib/nested.sh
+    . "$TESTSLIB"/nested.sh
+
+    variant=18
+    if os.query is-focal; then
+        variant=20
+    fi
+
+    boot_id="$(tests.nested boot-id)"
+
+    tests.nested exec "snap model --assertion" | MATCH "brand-id: $SNAPD_TEST_BRAND\$"
+    tests.nested exec "snap model --assertion" | MATCH "store: $SNAPD_STORE_WHITEBOX\$"
+    tests.nested exec "snap model --assertion" | MATCH '^model: test-snapd-remodel-pc$'
+    tests.nested exec "snap model --assertion --serial" | MATCH "serial: ${SERIAL}\$"
+
+    # the new model requires test-snapd-tools-core* snap to be present, make
+    # sure that the snap isn't installed yet
+    not tests.nested exec "snap list" | NOMATCH test-snapd-tools
```

Comment:
Why do we have a `not` here? Do we expect `snap list` to fail?

Reply:
ah, good catch